test_check_dvcs.py

```diff
@@ -234,128 +243,84 @@ def test_failed_check(self):
+                None,
+                True,
             ),
-            (  # Validate AAP-1234 same marker format
-                'AAP-1234 this is a title',
-                ['AAP-1234', 'aap-1234', 'aAp-1234'],
-                'aap-1234 this is the source branch',
-                f"* {check_dvcs.bad_icon} Mismatch: The JIRAs in the source branch",
+            (  # Key only in commit
+                None,
+                ['AAP-93993'],
+                None,
+                True,
             ),
-            (  # Validate AAP-1234 marker format
-                'AAP-3939 this is a title',
-                ['AAP-3939', 'AAP-3939', 'AAP-3939'],
-                'AAP-3939 this is the source branch',
-                f"* {check_dvcs.bad_icon} Mismatch: No commit with source branch JIRA number",
+            (  # Key only in PR title
+                None,
+                [],
+                'AAP-77888',
+                True,
             ),
-            (  # Commit: no commits with a Jira number
-                'AAP-4545 this is a title',
+            (  # No key in PR title, branch name, or commit.
+                None,
                 [],
-                'AAP-4545 this is the source branch',
-                f"* {check_dvcs.bad_icon} Commits: No commits with a JIRA number (AAP-[0-9]+) or NO_JIRA found!"
+                None,
+                False,
             ),
         ],
         ids=[
-            "PR title is none",
-            "PR title does not match expected format",
-            "Commit does not match pr title",
-            "Commit JIRA markers don't match PR and SB JIRA markers",
-            "_NO_JIRA title does not care about anything else",
-            "Source branch is none",
+            "No key in PR title, commits and source branch both have NO_JIRA",
+            "Key in PR title, multiple in commits, key in branch name",
+            "NO_JIRA in PR title, keys in commits and source branch",
+            "Key in PR title and commit, not in branch name",
             "Source branch does not match jira PR",
-            "Source branch does not match expected format",
-            "Validate low and upper case AAP-1234 marker format",
-            "Validate AAP-1234 same marker format",
-            "Validate AAP-1234 marker format",
-            "Commit: no commits with a Jira number"
+            "Key in PR title and branch name, not in commit",
+            "Key only in PR title",
+            "Key only in commit",
+            "Key only in PR title",
+            "No key in PR title, branch name, or commit.",
         ],
     )
-    def test_decissions_output(self, pr_title_jira, possible_commit_jiras, source_branch_jira, expected_in_message):
-        result = check_dvcs.make_decisions(pr_title_jira, possible_commit_jiras, source_branch_jira)
-        assert expected_in_message in result
+    def test_decisions_output(self, pr_title_jira, possible_commit_jiras, source_branch_jira, expected_result):
```

Comment:
can we add test cases for:
- JIRA ticket in the middle of a string
- jIrA ticket with wrong case